polkadot/node/core/candidate-validation/src/lib.rs

```diff
@@ -1030,17 +1058,34 @@ async fn validate_candidate_exhaustive(
 								return Err(ValidationFailed(error.into()))
 							};
 
-							if let Err(err) = committed_candidate_receipt
-								.check_core_index(&transpose_claim_queue(claim_queue.0))
-							{
+							let Some(node_features) = maybe_node_features else {
```

Comment:
Why do we need the node features ? We can simply just check all of them without checking the feature. We can rely on collator protocol and statement distribution to do these first level checks.

Reply:
collator protocol does not check the commitments because it doesn't have them (they're outputs of execution)